src/Numerics/Mesh/BrickMesh.jl

```diff
@@ -1126,4 +1126,341 @@ function enumerateboundaryfaces!(elemtoelem, elemtobndy, periodicity, boundary)
+        for f in 1:nfaces
+            j = (e - 1) * nfaces + f
+            fv, o = vertsortandorder(v[fmask[:, f]]...) # faces vertices, B-N orientation
+            for fvt in 1:nfvert
+                A[fvt, j] = gldofv[fv[fvt]]
+            end
+            A[nfvert + 1, j] = o # orientation
+            A[nfvert + 2, j] = e # local element number
+            A[nfvert + 3, j] = f # local face number for element e
+        end
+    end
+    A = sortslices(A, dims = 2, by = x -> x[1:nfvert])
+    elemtoelem = Array{Int}(undef, nfaces, (nelem + nghost))
+    elemtoface = Array{Int}(undef, nfaces, (nelem + nghost))
+    elemtoordr = Array{Int}(undef, nfaces, (nelem + nghost))
+    # match faces
+    n = size(A, 2)
+    j = 1
+    while j ≤ n
+        lel = A[nfvert + 2, j]
+        lfc = A[nfvert + 3, j]
+        if j + 1 ≤ n && A[1:nfvert, j] == A[1:nfvert, j + 1]
+            # found connected face
+            nel = A[nfvert + 2, j + 1] # local neighboring element
+            nfc = A[nfvert + 3, j + 1] # local face # of local neighboring element
+            elemtoelem[lfc, lel] = nel
+            elemtoface[lfc, lel] = nfc
+            elemtoelem[nfc, nel] = lel
+            elemtoface[nfc, nel] = lfc
+            if A[nfvert + 1, j] == A[nfvert + 1, j + 1]
+                elemtoordr[lfc, lel] = 1
+                elemtoordr[nfc, nel] = 1
+            else
+                elemtoordr[lfc, lel] = 2
+                elemtoordr[nfc, nel] = 2
+            end
+            j += 2
+        else
+            # found unconnected face
+            elemtoelem[lfc, lel] = lel
+            elemtoface[lfc, lel] = lfc
+            elemtoordr[lfc, lel] = 1
+            j += 1
+        end
+    end
+    #---------------------------------------------------------------
+    (
+        elems = 1:(nelem + nghost),       # range of element indices
+        realelems = 1:nelem,              # range of real element indices
+        ghostelems = nelem .+ (1:nghost), # range of ghost element indices
+        ghostfaces = ghostfaces,          # bit array of recv faces for ghost elems
+        sendelems = newsendelems,         # array of send element indices
+        sendfaces = sendfaces,            # bit array of send faces for send elems
+        elemtocoord = newelemtocoord,     # element to vertex coordinates
+        elemtoelem = elemtoelem,          # element to neighboring element
+        elemtoface = elemtoface,          # element to neighboring element face
+        elemtoordr = elemtoordr,          # element to neighboring element order
+        elemtobndy = newelemtobndy,       # element to boundary number
+        nabrtorank = nabrtorank,          # list of neighboring processes MPI ranks
+        nabrtorecv = nabrtorecv,          # neighbor receive ranges into `ghostelems`
+        nabrtosend = nabrtosend,          # neighbor send ranges into `sendelems`
+    )
+end
+
+function set_fmask(dim, nvert, nfaces)
```

Comment:
what does this do exactly? is it a remapping of nodes? If so, can it be coded in more straighforward way or have a clear explanation of what this is doing?

Reply:
This obtains face vertices from cell/element vertices

Reply:
can we keep the code as readable as possible rather than using a lot of julia "shortcuts" to achieve the same result?

Reply:
Should have a docstring. Also, isn't `nvert` redundant here, since `nvert = 2^dim`?

Reply:
Also `nfaces = 2dim`?

Reply:
and maybe rename to `build_fmask`